lib/ansible/modules/pip.py

```diff
@@ -611,6 +613,64 @@ def setup_virtualenv(module, env, chdir, out, err):
     return out, err, cmd
 
 
+def _resolve_package_names(
+        module: AnsibleModule,
+        package_list: list[Package],
+        pip: list[str]
+) -> list[Package]:
+    """Return a list of Packages with names extracted from metadata using pip"""
+    pkgs_to_resolve = [pkg for pkg in package_list if not pkg.has_requirement]
+
+    if not pkgs_to_resolve:
+        return package_list
+
+    # Unsetting 'FORCE_COLOR' with 'NO_COLOR' and 'TTY_COMPATIBLE' env vars, helps to
+    # make pip output clean JSON without injected ANSI sequences that might corrupt it.
+    os.environ.pop('FORCE_COLOR', None)
+
+    rc, json_out, err = module.run_command(
+        [*pip, 'install', '--dry-run', '--ignore-installed', '--quiet', '--report=-', *(str(pkg) for pkg in pkgs_to_resolve)],
```

Comment:
I think this might be more elegant:
```suggestion
        (*pip, 'install', '--dry-run', '--ignore-installed', '--quiet', '--report=-', *map(str, pkgs_to_resolve)),
```
FWIW, having nested expressions always increases cognitive complexity. And having that inline also increases the https://github.com/Miserlou/JonesComplexity score. You might want to put that into a named var.

Reply:
I don't think that making it a map instead of a comprehension makes it significantly clearer so I'm leaving as is